scripts/py_matter_idl/matter_idl/data_model_xml/handlers/parsing.py

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import re
+from typing import Optional
+
+from matter_idl.generators.types import GetDataTypeSizeInBits, IsSignedDataType
+from matter_idl.matter_idl_types import (AccessPrivilege, Attribute, AttributeQuality, Command, ConstantEntry, DataType, Event,
+                                         EventPriority, Field, FieldQuality)
+
+LOGGER = logging.getLogger('data-model-xml-data-parsing')
+
+
+def ParseInt(value: str, data_type: Optional[DataType] = None) -> int:
+    """
+    Convert a string that is a known integer into an actual number.
+
+    Supports decimal or hex values prefixed with '0x'
+    """
+    if value.startswith('0x'):
+        parsed = int(value[2:], 16)
+        if data_type and IsSignedDataType(data_type):
+            bits = GetDataTypeSizeInBits(data_type)
+            assert (bits)  # size MUST be known
+            if parsed & (1 << (bits - 1)):
+                parsed -= 1 << bits
+        return parsed
+    else:
+        return int(value)
+
+
+def ParseOptionalInt(value: str) -> Optional[int]:
+    """Parses numbers as long as they are in an expected format of numbers.
+
+       "1" parses to 1
+       "0x12" parses to 18
+       "Min" parses to None
+    """
+    if re.match("^-?((0x[0-9a-fA-F]*)|([0-9]*))$", value):
+        return ParseInt(value)
+
+    return None
+
+
+_TYPE_REMAP = {
```

Comment:
I would consider it if we can gather where things happen. We have some type logic in generators, but those use the 2nd value from here (i.e. the idl type) and the remap that zap does is probably inside zap javascript so hard to share.

My belief is that we do have several mappings, but every one may be ever so slightly different :(

Reply:
Closing since I am unsure how likely we are to consolidate these for now and I am only aware of different usages that cannot overlap. If we build up more tooling, we can start sharing. For now in matter_idl itself this map is unique.